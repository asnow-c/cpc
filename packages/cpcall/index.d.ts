/**
 * api抽取
 */

import * as core from "./out/core/cpc.js";
import * as node from "./out/node/index.js";
import * as web from "./out/web/index.js";
export { core, web, node };
