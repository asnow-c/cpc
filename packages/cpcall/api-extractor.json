{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",

    "projectFolder": ".",
    "extends": "../../api-extractor.jsonc",
    "mainEntryPointFilePath": "<projectFolder>/index.d.ts",
    "compiler": {
        "tsconfigFilePath": "<projectFolder>/tsconfig.json",
        "overrideTsconfig": {
            "$schema": "http://json.schemastore.org/tsconfig",
            "compilerOptions": {
                "target": "ES2022",
                "module": "NodeNext",
                "moduleResolution": "nodenext",
                "skipDefaultLibCheck": true,
                "strict": true,
                "lib": ["ESNext"],
                "paths": {
                    "cpcall/node": ["./dist/node/index.js"],
                    "cpcall/web": ["./dist/web/index.js"],
                    "cpcall": ["./dist/cpc/cpc.js"],
                    "#lib/*": ["./dist/cpc/lib/*"]
                }
            },
            "files": ["index.d.ts"],
            "exclude": ["**/*.ts"]
        }
    },
    "messages": {
        "tsdocMessageReporting": {
            "tsdoc-escape-right-brace": {
                "logLevel": "none"
            },
            "tsdoc-malformed-inline-tag": {
                "logLevel": "none"
            }
        }
    }
}
