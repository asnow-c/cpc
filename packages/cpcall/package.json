{
    "name": "cpcall",
    "version": "0.0.2",
    "description": "",
    "author": {
        "name": "Eavid",
        "url": "https://github.com/ieavid"
    },
    "type": "module",
    "scripts": {
        "ci:build": "tsc -p tsconfig.build.json",
        "publish:test": "pnpm publish --no-git-checks --dry-run",
        "api:extractor": "api-extractor run -v",
        "api:extractor-local": "pnpm run api:extractor -lc api-extractor.json"
    },
    "devDependencies": {
        "@microsoft/api-extractor": "^7.36.4",
        "@types/node": "^20.4.4"
    },
    "exports": {
        ".": "./out/core/cpc.js",
        "./node": "./out/node/index.js",
        "./web": "./out/web/index.js"
    },
    "imports": {
        "#lib/*": "./out/core/lib/*"
    },
    "publishConfig": {
        "access": "public",
        "registry": "https://registry.npmjs.org"
    },
    "files": [
        "**/out/**/*.js",
        "**/out/**/*d.ts"
    ],
    "repository": {
        "type": "git",
        "url": "https://github.com/ieavid/cpcall"
    },
    "dependencies": {
        "@eavid/js-bson": "workspace:^"
    }
}
