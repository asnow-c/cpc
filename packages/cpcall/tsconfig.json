{
    "extends": "../../tsconfig.json",
    "compilerOptions": {
        "lib": ["ESNext", "DOM"],
        "baseUrl": ".",
        "rootDir": ".",
        "paths": {
            "cpcall/node": ["./src/node/index.js"],
            "cpcall/web": ["./src/web/index.js"],
            "cpcall": ["./src/cpc/cpc.js"],
            "#lib/*": ["./src/cpc/lib/*"]
        }
    },
    "exclude": ["node_modules", "out"]
}
