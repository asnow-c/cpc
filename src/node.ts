export * from "./node/node_ps_cpc.js";
export * from "./node/socket_cpc.js";
