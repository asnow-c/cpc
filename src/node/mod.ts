export * from "./socket_cpc.js";
