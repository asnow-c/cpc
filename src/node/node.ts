export * from "./node_ps_cpc.js";
export * from "./socket_cpc.js";
