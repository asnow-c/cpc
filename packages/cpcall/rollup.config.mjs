import * as rollup from "@eavid/lib-dev/rollup";
import * as path from "node:path";

/** @type {typeof import("@eavid/lib-dev/out/rollup/mod")} */
const { defineEvConfig } = rollup
const dir = process.cwd()
console.log("cwd:" + dir);
const libDir = path.resolve(dir, "src/cpc/lib")

export default defineEvConfig({
    input: {
        "cpc": "src/cpc/cpc.ts",
        "node": "src/node/node.ts",
        "web": "src/web/web.ts"
    },
    output: {
        dir: "dist",
        chunkFileNames: "internal/[name].js",
        minifyInternalExports: false,
        entryFileNames: "[name]/[name].js",
        manualChunks(id, meta) {
            if (id.startsWith(libDir)) return "lib"
        }
    },
    extra: {
        typescript: {
            module: "NodeNext",
            rootDir: "src",
            outDir: "./dist",
            noEmit: false,
            declaration: true,
            declarationMap: true
        }
    }
});