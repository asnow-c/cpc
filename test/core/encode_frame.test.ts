import { RpcFrame, FrameType, encodeCpcFrame } from "cpcall";
import { describe } from "node:test";
import { expect, test } from "vitest";
import { baseDataTypes } from "../__mocks__/data_type.cases.js";
function encodeCpcFrameHex(frame: RpcFrame) {
  const u8Arr = encodeCpcFrame(frame);
  return u8ArrToHex(u8Arr);
}
describe("caller", function () {
  test("call", async function () {
    const hex = encodeCpcFrameHex([FrameType.call, ["abcd", 1, "null"]]);
    expect(hex).toMatchSnapshot();
  });
  test("exec", async function () {
    const hex = encodeCpcFrameHex([FrameType.exec, ["abcd", 1, "null"]]);
    expect(hex).toMatchSnapshot();
  });
});
describe("response", function () {
  test("return", async function () {
    const hex = encodeCpcFrameHex([FrameType.return, 3]);
    expect(hex).toMatchSnapshot();
  });
  test("throw", async function () {
    const hex = encodeCpcFrameHex([FrameType.throw, new Error("err")]);
    expect(hex).toMatchSnapshot();
  });
  test("promise", async function () {
    const hex = encodeCpcFrameHex([FrameType.promise, 325]);
    expect(hex).toMatchSnapshot();
  });
  test("resolve", async function () {
    const hex = encodeCpcFrameHex([FrameType.resolve, 3, "ab"]);
    expect(hex).toMatchSnapshot();
  });
  test("reject", async function () {
    const hex = encodeCpcFrameHex([FrameType.reject, 3, "ab"]);
    expect(hex).toMatchSnapshot();
  });
});
describe("action", function () {
  test("fin", async function () {
    const hex = encodeCpcFrameHex([FrameType.end]);
    expect(hex).toMatchSnapshot();
  });
  test("disable", async function () {
    const hex = encodeCpcFrameHex([FrameType.disable]);
    expect(hex).toMatchSnapshot();
  });
});
/** 测试参数传输 */
describe("数据类型二进制", function () {
  const cases = Object.entries(baseDataTypes);
  test.each(cases)("%s", async function ([key, arg]) {
    const hex = encodeCpcFrameHex([FrameType.call, ["fn", ...arg]]);
    expect(hex).toMatchSnapshot();
  });
});

function u8ArrToHex(u8Arr: Uint8Array) {
  return Buffer.from(u8Arr, u8Arr.byteOffset, u8Arr.byteLength).toString("hex");
}
