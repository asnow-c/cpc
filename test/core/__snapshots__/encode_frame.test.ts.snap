// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`array 1`] = `"010a02666e0a017200"`;

exports[`bigint 1`] = `"010a02666e0a016900"`;

exports[`call 1`] = `"010a046162636414000000010a046e756c6c00"`;

exports[`disable 1`] = `"ff"`;

exports[`double 1`] = `"010a02666e0a016f00"`;

exports[`error 1`] = `"010a02666e0a017200"`;

exports[`exec 1`] = `"020a046162636414000000010a046e756c6c00"`;

exports[`fin 1`] = `"fe"`;

exports[`int 1`] = `"010a02666e0a016e00"`;

exports[`map 1`] = `"010a02666e0a016100"`;

exports[`noContent 1`] = `"010a02666e0a016f00"`;

exports[`promise 1`] = `"08c502"`;

exports[`regExp 1`] = `"010a02666e0a016500"`;

exports[`reject 1`] = `"0a030a026162"`;

exports[`resolve 1`] = `"09030a026162"`;

exports[`return 1`] = `"0b1400000003"`;

exports[`string 1`] = `"010a02666e0a017400"`;

exports[`symbol 1`] = `"010a02666e0a017900"`;

exports[`throw 1`] = `"0c200a076d657373616765036572720a046e616d65054572726f7200"`;
